elfo-network/src/socket/handshake.rs

```diff
@@ -39,6 +41,18 @@ impl Handshake {
         }
     }
 
+    pub(super) fn choose_compression(&self) -> Option<CompressionAlgorithm> {
```

Comment:
I'm sure all helpers should be defined on `capabilities`. Otherwise, any new capabilities will also require an additional method here, but it's pointless while `Capabilities` is reachable directly (because this is a public field of `Handshake`).